ldap/servers/slapd/log.c

```diff
@@ -225,63 +218,47 @@ slapd_log_error_proc_internal(
-    do {                                                                                              \
-        (err) = 0;                                                                                    \
-        if (slapi_write_buffer((fd), (buffer), (PRInt32)(size)) != (PRInt32)(size)) {                 \
-            PRErrorCode prerr = PR_GetError();                                                        \
-            syslog(LOG_ERR, "Failed to write log, " SLAPI_COMPONENT_NAME_NSPR " error %d (%s): %s\n", \
-                   prerr, slapd_pr_strerror(prerr), (buffer) + (headersize));                         \
-            (err) = prerr;                                                                            \
-            rc = -1;                                                                                  \
-        }                                                                                             \
-        /* Should be a flush in here ?? Yes because PR_SYNC doesn't work ! */                         \
-        PR_Sync(fd);                                                                                  \
-    } while (0)
-#define LOG_WRITE_NOW_NO_ERR(fd, buffer, size, headersize)                                            \
-    do {                                                                                              \
-        if (slapi_write_buffer((fd), (buffer), (PRInt32)(size)) != (PRInt32)(size)) {                 \
-            PRErrorCode prerr = PR_GetError();                                                        \
-            syslog(LOG_ERR, "Failed to write log, " SLAPI_COMPONENT_NAME_NSPR " error %d (%s): %s\n", \
-                   prerr, slapd_pr_strerror(prerr), (buffer) + (headersize));                         \
-            rc = -1;                                                                                  \
-        }                                                                                             \
-        /* Should be a flush in here ?? Yes because PR_SYNC doesn't work ! */                         \
-        PR_Sync(fd);                                                                                  \
-    } while (0)
-#define LOG_CLOSE(fd) \
-    PR_Close((fd))
+static int32_t
+log_write(LOGFD fd, char *buffer, int32_t size, int32_t headersize)
+{
+    int32_t rc = 0;
+
+    if (slapi_write_buffer((fd), (buffer), (PRInt32)(size)) != (PRInt32)(size)) {
+        PRErrorCode prerr = PR_GetError();
+        syslog(LOG_ERR, "Failed to write log, " SLAPI_COMPONENT_NAME_NSPR " error %d (%s): %s\n",
+               prerr, slapd_pr_strerror(prerr), (buffer) + (headersize));
+        rc = -1;
+    }
+
+    return rc;
+}
+
+/*
+ * Writes into a LOGFD and flushes the buffer if necessary
+ */
+static int32_t
+log_write_now(LOGFD fd, char *buffer, int32_t size, int32_t headersize)
```

Comment:
Instead of two functions, we could only use log_write and add a flag 'flush' to Pr_sync.

Reply:
> Instead of two functions, we could only use log_write and add a flag 'flush' to Pr_sync.

Sure I can do that

Reply:
> Instead of two functions, we could only use log_write and add a flag 'flush' to Pr_sync.

Done!